crates/polars-lazy/src/frame/mod.rs

```diff
@@ -1138,6 +1138,18 @@ impl LazyFrame {
         if let Expr::Column(name) = index_column {
             options.index_column = name;
         } else {
+            fn is_int_range(expr: &Expr) -> bool {
+                match expr {
+                    Expr::Alias(input, _) => is_int_range(input),
+                    Expr::Function {
+                        input, function, ..
+                    } => {
+                        matches!(function, FunctionExpr::Range(f) if f.to_string() == "int_range")
```

Comment:
A bit awkward that I have to check against the `.to_string` method.

Reply:
I think we should do this check during the IR::conversion, in `resolve_groupby` here:

https://github.com/pola-rs/polars/blob/05f2abbf1b7f76f0b34c3c552fc33aa6da186561/crates/polars-plan/src/plans/conversion/dsl_to_ir.rs#L1013

Reply:
Sure! I am just unsure how I should check in `resolve_group_by` whether the index of the group_by_dynamic is an int_range. Can I get the expression for this index (conveniently) from the `expr_arena`?

Reply:
Ah, right. I see that we first do a `with_columns` here. We should store the `index` column on line 1155 and do the `with_column` rewrite during IR conversion.

I understand it's a bit more than you anticipated. I can do the pre-work for that later if you like?

Reply:
I suppose this involves passing the index_column `Expr` from the `lazy_frame.group_by_dynamic` method to the dsl_to_ir functions? However, the best approach to do this is not clear to me..

Would thus be very helpful if you could do the pre-work for this! :)